### ISSUE TITLE
Configuring for AWS - what's the format of the `index_name` parameter?

### ISSUE THREAD

**Issue by leesmith** [alexreisner/geocoder#1653]
I'm trying to configure Geocoder for AWS. I've created what I think is a places index in AWS and assigned an api key to it. I tried to use the full ARN value of the index for `index_name` in the `amazon_location_service` config block but that doesn't work:

> .asdf/installs/ruby/3.3.0/lib/ruby/gems/3.3.0/gems/aws-sdk-core-3.191.6/lib/seahorse/client/plugins/raise_response_errors.rb:17:in `call': 'IndexName' must match regex ^[-._\w]+$ (Aws::LocationService::Errors::ValidationException)

I also tried just the name that I gave it but that didn't work either...so what is the format of the `index_name` config parameter? Thanks!

* Geocoder version: 1.8.2
* Rails version: 7.1.3.2
* Database (if applicable): 
* Lookup (if applicable): amazon_location_service

**Comment by alexreisner** [comment]
Hmm, what's the name of your index? It looks like it can only have letters, digits, dash, underscore, and period (so an ARN wouldn't work).

**Comment by leesmith** [comment]
Thanks for the reply!

So I think my issue was kind of a combo of errors. I configured `index_name` to just be the index name without any invalid characters (not the ARN value).

After that, I got an invalid key error. In AWS location services, there's a place to create an API key but I don't think this is what's actually needed. I then deleted that API key and went into IAM and created a user with full "geo" permissions as [outlined here](https://docs.aws.amazon.com/location/latest/developerguide/gs-prereqs.html#gs-grant-access) and now it works.
